pkg/controller/replica_test.go

```diff
@@ -68,7 +69,7 @@ func TestReplicaReconcile(t *testing.T) {
 				replicaAttachment := testReplicaAzVolumeAttachment
 				now := metav1.Time{Time: metav1.Now().Add(-1000)}
 				replicaAttachment.DeletionTimestamp = &now
-				replicaAttachment.Status.State = azdiskv1beta2.Detaching
+				replicaAttachment.Status.State = azdiskv1beta2.DetachmentFailed
```

Comment:
Let's remove this line for this test case, attachment's state should not have any impact on this test case and will only confuse the readers going forward.

Instead, consider adding a new test case "[Success] Should delete replica attachment if attachment fails and requeue" to ensure that the reconciler will delete replica attachment in `AttachmentFailed` state and return reconcile result, where `requeue == true`

Reply:
>Instead, consider adding a new test case "[Success] Should delete replica attachment if attachment fails and requeue" to ensure that the reconciler will delete replica attachment in AttachmentFailed state and return reconcile result, where requeue == true

Oh, you have already added "[Success] Should delete a failed-attachment replica." Let's go ahead and remove this line in that case 👍

Reply:
I edited the attachment's state to `DetachmentFailed` for testing [this case](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/fde668a0d77effb4af845a115572180f85a65f9f/pkg/controller/replica.go#L95). And we can verify if the state has been updated to `ForceDetachPending` right before [delete the attachment manually in the verifyFun](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/fde668a0d77effb4af845a115572180f85a65f9f/pkg/controller/replica_test.go#L96). Not sure if it's needed?

Reply:
For this test case, we want to check if a new replica attachment is created when one is deleted.

Reply:
Got it. I've updated the code.